pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/InstanceCache.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.functions.instance;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class InstanceCache {
```

Comment:
what is the consideration behind this `InstanceCache` class? because it seems to be only used in JavaInstanceRunnable. Then why not just run it as part of JavaInstanceRunnable.

Reply:
This was done in consideration to ThreadedRuntime.  If running with ThreadRuntime, java instances can just share a cache and the timer thread.  While in other runtimes each instance will have its own cache and timer thread.